[Clang] Fix a crash introduced in PR#88666

shiltian opened the pull request:
The unroll value can be a template variable such that we need to check it before
we verify if it is constant value.

yronglin reviewed (approved):
Thanks for your fix, LGTM!

yronglin commented on clang/test/Sema/unroll-template-value-crash.cpp:
```diff
@@ -0,0 +1,7 @@
+template <int Unroll> void foo() {
+  #pragma unroll Unroll
+  for (int i = 0; i < Unroll; ++i);
+}
```
I'd like to add a test for '#pragma GCC unroll N', WDYT?

shiltian committed:
[Clang] Fix a crash introduced in PR#88666

The unroll value can be a template variable such that we need to check it before
we verify if it is constant value.
